receiver/snmpreceiver/scraper.go

```diff
@@ -39,13 +66,364 @@ func newScraper(logger *zap.Logger, cfg *Config, settings component.ReceiverCrea
+		var attributeValue string
+		if prefix := configHelper.getResourceAttributeConfigIndexedValuePrefix(attributeName); prefix != "" {
+			attributeValue = prefix + indexString
+		}
+		if oid := configHelper.getResourceAttributeConfigOID(attributeName); oid != "" {
+			indexedAttrKey := indexedAttributeKey{
+				parentOID: oid,
+				oidIndex:  indexString,
+			}
+			attributeValue = indexedResourceAttributeValues[indexedAttrKey]
+		}
+		if attributeValue == "" {
+			return nil, errors.New(errMsgResourceAttributeEmptyValue)
+		}
+		resourceAttributes[attributeName] = attributeValue
+	}
+
+	return resourceAttributes, nil
+}
+
+// scrapeIndexedAttributes retrieves all SNMP data from attribute config column OIDs and
+// stores the returned indexed data for later use by metrics
+func (s *snmpScraper) scrapeIndexedAttributes(
+	configHelper *configHelper,
+	scraperErrors *scrapererror.ScrapeErrors,
+) map[indexedAttributeKey]string {
+	indexedAttributeValues := map[indexedAttributeKey]string{}
+
+	// Find all attribute column OIDs
+	attributeColumnOIDs := configHelper.getAttributeColumnOIDs()
+
+	// If no oid attribute configs, nothing else to do
+	if len(attributeColumnOIDs) == 0 {
+		return indexedAttributeValues
+	}
+
+	// Retrieve all SNMP indexed data from column attribute OIDs
+	indexedData := s.client.GetIndexedData(attributeColumnOIDs, scraperErrors)
+	// For each piece of SNMP data, store the necessary info to help create metric attributes later if needed
+	for _, data := range indexedData {
+		if err := indexedDataToAttribute(data, indexedAttributeValues); err != nil {
+			scraperErrors.AddPartial(1, fmt.Errorf(errMsgIndexedAttributeOIDProcessing, data.oid, data.parentOID, err))
+		}
+	}
+
+	return indexedAttributeValues
+}
+
+// scrapeIndexedResourceAttributes retrieves all SNMP data from resource attribute config column OIDs and
+// stores the returned indexed data for later use by metrics
+func (s *snmpScraper) scrapeIndexedResourceAttributes(
+	configHelper *configHelper,
+	scraperErrors *scrapererror.ScrapeErrors,
+) map[indexedAttributeKey]string {
+	indexedResourceAttributeValues := map[indexedAttributeKey]string{}
+
+	// Find all resource attribute column OIDs
+	resourceAttributeColumnOIDs := configHelper.getResourceAttributeColumnOIDs()
+
+	// If no OID resource attribute configs, nothing else to do
+	if len(resourceAttributeColumnOIDs) == 0 {
+		return indexedResourceAttributeValues
+	}
+
+	// Retrieve all SNMP indexed data from column resource attribute OIDs
+	indexedData := s.client.GetIndexedData(resourceAttributeColumnOIDs, scraperErrors)
+	// For each piece of SNMP data, store the necessary info to help create resources later if needed
+	for _, data := range indexedData {
+		if err := indexedDataToAttribute(data, indexedResourceAttributeValues); err != nil {
+			scraperErrors.AddPartial(1, fmt.Errorf(errMsgIndexedResourceAttributeOIDProcessing, data.oid, data.parentOID, err))
+		}
+	}
+
+	return indexedResourceAttributeValues
+}
```

Comment:
These functions are nearly identical. Please combine and pass in the appropriate set of ColumnOIDs.

Reply:
@djaglowski so i could do this. but then i'd have to do one of two things.
either also pass in some parameter so i can determine which error message to use in the loop (bool or maybe the error string itself)
or i could shift the final loop to be outside of the newly combined function. and this way i could keep this custom to attribute or resource attribute processing. this would be a little unfortunate as it wouldn't match up as well with the other scrape functions.  actually i can still keep these and only combine the like parts